### ISSUE TITLE
The final `{*opts}` parameter, specified for many of the Lua Nvim Api calls should be optional in use.

### ISSUE THREAD

**Issue by strait** [neovim/neovim#18882]
### Feature already in Vim?

no

### Feature description

It is customary in Lua to leave out the final optional table argument, if not used. The parameter then takes the value `nil`, indicating the default. The API strictly requiring a `{}` adds no additional information.

If you need additional options, then you add them. If not, then expect the defaults, which are documented. The additional syntax clunk is not necessary.

**Comment by clason** [comment]
The Neovim API (`nvim_XXX`) is not just for Lua, though, and other supported languages are much more strict about that.

**Comment by bfredl** [comment]
https://github.com/neovim/neovim/pull/15302 this has already been discussed enough.

**Comment by strait** [comment]
Okay, I hadn't found that duplicate in my search. Someone mentioned the potential of having an additional parameter added beyond the optional table, which would never happen in practice.

I understand that `nvim_XXX` has additional clients beyond Lua, but unnecessarily forcing the common denominator on Lua isn't a good design decision.

Likewise, I noticed straight away that all `vim.api.nvim_` API calls include a redundancy in the naming.
The first thing I did was add a `Va = setmetatable(vim.api, {__index = function(t k) return rawget(t, "nvim_" .. k) end})` to my Lua code. So, `vim.api.nvim_create_autocmd` can now be called as `Va.create_autocmd`
Some developers will use the official API naming and others of us will want something more immediately comprehensible. Thus begins the additional metatable lookups, continual naming translations, and divergence.
